lib/spack/spack/test/conftest.py

```diff
@@ -1764,3 +1764,8 @@ def _spider(*args, **kwargs):
     monkeypatch.setattr(spack.util.web, "spider", _spider)
 
     yield
+
+
+@pytest.fixture(scope="function")
+def mock_tty_stdout(monkeypatch):
+    monkeypatch.setattr(sys.stdout, "isatty", lambda: True)
```

Comment:
`test_url_with_status_bar()` in `test/url_fetch.py` *should* work with this too, but the test fails with that for some reason. It's the only other place that `isatty` is patched in our testing so I think it's not worth figuring out why.